### PR TITLE
add test for #711

### PR DESCRIPTION
**Pull request by ericpyle**
passing test for #711 (originally from my issue #657)

### CODE REVIEW

File: test/MUIDataTable.test.js [hunk 1]
```diff
@@ -429,6 +430,24 @@ describe('<MUIDataTable />', function() {
...
+  it('should apply columns prop change for filterList', () => {
+    const mountShallowWrapper = mount(shallow(<MUIDataTable columns={columns} data={data} />).get(0));
+    const instance = mountShallowWrapper.instance();
+    instance.initializeTable(mountShallowWrapper.props());
+    // now use updated columns props
+    const newColumns = cloneDeep(columns);
+    newColumns[0].options.filterList = ['Joe James'];
+    mountShallowWrapper.setProps({ columns: newColumns });
+    mountShallowWrapper.update();
+    // mimic componentDidUpdate(prevProps) {
```
Comment (gabrielliwerant) [inline test/MUIDataTable.test.js]: I think we can remove these two comments. My reasoning is that they will be brittle; if the actual code changes, the comments will be out of date, and could lead to confusion.

File: test/MUIDataTable.test.js [hunk 2]
```diff
@@ -429,6 +430,24 @@ describe('<MUIDataTable />', function() {
...
+    // if (this.props.data !== prevProps.data || this.props.columns !== prevProps.columns) {
+    instance.setTableData(mountShallowWrapper.props(), 1 /* instance.TABLE_LOAD.INITIAL */);
+
+    const updatedState = mountShallowWrapper.state();
+    const { columns: updatedColumns } = updatedState;
```
Comment (gabrielliwerant) [inline test/MUIDataTable.test.js]: This line doesn't look like it's doing anything in the test, so I think we can remove it.

File: test/MUIDataTable.test.js (hunk 1 above)
Comment (ericpyle) [reply test/MUIDataTable.test.js]: Sounds good to me.

File: test/MUIDataTable.test.js (hunk 2 above)
Comment (ericpyle) [reply test/MUIDataTable.test.js]: You're right. Thanks.

**Comment by ericpyle** [comment]
@gabrielliwerant I think I've made the changes you suggested

**Pull request merged**